src/config.rs

```diff
@@ -266,14 +276,30 @@ fn default_convert_dos_to_unix() -> bool {
 }
 
 impl Config {
-    pub fn from_file(path: impl AsRef<Path>) -> Result<Self> {
-        let contents = fs::read_to_string(path.as_ref()).context(ReadConfigFileSnafu {
-            filename: path.as_ref(),
+    pub fn from_path(path: &PathBuf) -> Result<Self> {
+        let filename = if path.is_dir() {
+            path.join("mujmap.toml")
+        } else {
+            path.clone()
+        };
+
+        let contents = fs::read_to_string(&filename).context(ReadConfigFileSnafu {
+            filename: filename.clone(),
         })?;
-        let config: Self = toml::from_str(contents.as_str()).context(ParseConfigFileSnafu {
-            filename: path.as_ref(),
+        let mut config: Self = toml::from_str(contents.as_str()).context(ParseConfigFileSnafu {
+            filename: filename.clone(),
         })?;
 
+        // In directory mode, if paths aren't offered then we use the config dir itself.
+        if path.is_dir() {
+            if config.mail_dir.is_none() {
+                config.mail_dir = Some(path.clone());
```

Comment:
We need to be very careful about relative paths here. The `-C` option is relative to the working directory, but paths in the config file are relative to the root of the maildir, so we *must* canonicalize `path` here before assigning it to `mail_dir` or `state_dir`. Then `Local` and `sync` only have to worry about paths relative to the mail root.

Reply:
Mmm, you're right - I hadn't really considered relative paths at all, so rarely do I use them outside of the shell.